homeassistant/components/nest/climate.py

```diff
@@ -1,19 +1,357 @@
-"""Support for Nest climate that dispatches between API versions."""
+"""Support for Google Nest SDM climate devices."""
+from __future__ import annotations
 
+from typing import Any, cast
+
+from google_nest_sdm.device import Device
+from google_nest_sdm.device_manager import DeviceManager
+from google_nest_sdm.device_traits import FanTrait, TemperatureTrait
+from google_nest_sdm.exceptions import ApiException
+from google_nest_sdm.thermostat_traits import (
+    ThermostatEcoTrait,
+    ThermostatHeatCoolTrait,
+    ThermostatHvacTrait,
+    ThermostatModeTrait,
+    ThermostatTemperatureSetpointTrait,
+)
+
+from homeassistant.components.climate import (
+    ATTR_HVAC_MODE,
+    ATTR_TARGET_TEMP_HIGH,
+    ATTR_TARGET_TEMP_LOW,
+    FAN_OFF,
+    FAN_ON,
+    PRESET_ECO,
+    PRESET_NONE,
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACAction,
+    HVACMode,
+)
 from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, UnitOfTemperature
 from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity import DeviceInfo
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
-from .climate_sdm import async_setup_sdm_entry
-from .const import DATA_SDM
-from .legacy.climate import async_setup_legacy_entry
+from .const import DATA_DEVICE_MANAGER, DOMAIN
+from .device_info import NestDeviceInfo
+
+# Mapping for sdm.devices.traits.ThermostatMode mode field
+THERMOSTAT_MODE_MAP: dict[str, HVACMode] = {
+    "OFF": HVACMode.OFF,
+    "HEAT": HVACMode.HEAT,
+    "COOL": HVACMode.COOL,
+    "HEATCOOL": HVACMode.HEAT_COOL,
+}
+THERMOSTAT_INV_MODE_MAP = {v: k for k, v in THERMOSTAT_MODE_MAP.items()}
+
+# Mode for sdm.devices.traits.ThermostatEco
+THERMOSTAT_ECO_MODE = "MANUAL_ECO"
+
+# Mapping for sdm.devices.traits.ThermostatHvac status field
+THERMOSTAT_HVAC_STATUS_MAP = {
+    "OFF": HVACAction.OFF,
+    "HEATING": HVACAction.HEATING,
+    "COOLING": HVACAction.COOLING,
+}
+
+THERMOSTAT_RANGE_MODES = [HVACMode.HEAT_COOL, HVACMode.AUTO]
+
+PRESET_MODE_MAP = {
+    "MANUAL_ECO": PRESET_ECO,
+    "OFF": PRESET_NONE,
+}
+PRESET_INV_MODE_MAP = {v: k for k, v in PRESET_MODE_MAP.items()}
+
+FAN_MODE_MAP = {
+    "ON": FAN_ON,
+    "OFF": FAN_OFF,
+}
+FAN_INV_MODE_MAP = {v: k for k, v in FAN_MODE_MAP.items()}
+FAN_INV_MODES = list(FAN_INV_MODE_MAP)
+
+MAX_FAN_DURATION = 43200  # 15 hours is the max in the SDM API
+MIN_TEMP = 10
+MAX_TEMP = 32
 
 
 async def async_setup_entry(
```

Comment:
Reviewer note: This file is a rename of `climate_sdm.py` with this method renamed.